src/components/utils/Menu/Menu.tsx

```diff
@@ -9,16 +9,39 @@ interface MenuProps {
   maxHeight: number;
 }
 
+function getOffset(el: HTMLElement) {
+  const rect = el.getBoundingClientRect();
+  return {
+    isLeft: rect.left < window.innerWidth / 2,
+    isBottom: rect.top + window.screenY > window.innerHeight / 2,
+  };
+}
+
 const Menu = React.forwardRef<HTMLElement, MenuProps>(function Menu(
   { children, isOpen, maxHeight, ...rest },
   ref
 ) {
+  const [positions, setPositions] = React.useState<{
+    top: string | number | null;
```

Comment:
Should we remove this variable and just store the right?

Reply:
I removed the `top`but I kept the object because I think it will be a fast followup